### PR TITLE
Move layer tree construction into flow::LayerBuilder so alternate builders can be plugged in.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This moves all the logic of construction of the layer tree into flow.

### CODE REVIEW

File: lib/ui/compositing/scene_builder.cc [hunk 1]
```diff
@@ -150,77 +108,35 @@ void SceneBuilder::pushShaderMask(Shader* shader,
...
-    m_currentLayer = m_rootLayer.get();
-    return;
-  }
-  if (!m_currentLayer)
-    return;
-  flow::ContainerLayer* newLayer = layer.get();
-  m_currentLayer->Add(std::move(layer));
-  m_currentLayer = newLayer;
+  layer_builder_.PushPhysicalModel(
+      rrect.sk_rrect,               //
```
Comment (jason-simmons) [inline lib/ui/compositing/scene_builder.cc]: Add text here or remove these empty comments

(a few other PushModel calls have similar formatting)

Comment (chinmaygarde) [reply lib/ui/compositing/scene_builder.cc]: I added those so clang format displayed them a separate line. I'll add comments.

**Comment by chinmaygarde** [comment]
Haven't tested this on Fuchsia yet. Will hold off on landing this till I do.

File: flow/layers/layer_builder.cc [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/backdrop_filter_layer.h"
+#include "flutter/flow/layers/clip_path_layer.h"
+#include "flutter/flow/layers/clip_rect_layer.h"
+#include "flutter/flow/layers/clip_rrect_layer.h"
+#include "flutter/flow/layers/color_filter_layer.h"
+#include "flutter/flow/layers/container_layer.h"
+#include "flutter/flow/layers/layer.h"
+#include "flutter/flow/layers/layer_builder.h"
```
Comment (abarth) [inline flow/layers/layer_builder.cc]: The primary header should be first, followed by a blank line, followed by the rest of the headers.  This pattern ensures that each header can build on its own (and doesn't require other headers to be included first).

**Comment by abarth** [comment]
LGTM

File: flow/layers/layer_builder.cc (hunk 2 above)
Comment (chinmaygarde) [reply flow/layers/layer_builder.cc]: Done.

**Pull request merged**